tests/test_processing_stack.py

```diff
@@ -220,17 +241,45 @@ def should_successfully_run_dataset_version_creation_process_with_multiple_asset
         ),
         bucket_name=Resource.STAGING_BUCKET_NAME.resource_name,
         key=f"{key_prefix}/{item_metadata_filename}",
-    ), Dataset() as dataset:
+    ):
 
         # When
         try:
+
+            dataset_response = lambda_client.invoke(
+                FunctionName=Resource.DATASETS_ENDPOINT_FUNCTION_NAME.resource_name,
+                Payload=dumps(
+                    {
+                        HTTP_METHOD_KEY: "POST",
+                        BODY_KEY: {
+                            TITLE_KEY: dataset_title,
+                            DESCRIPTION_KEY: any_dataset_description(),
+                        },
+                    }
+                ).encode(),
+            )
+            dataset_payload = load(dataset_response["Payload"])
+
+            with subtests.test(msg="Dataset endpoint returns success"):
```

Comment:
A subtest means we want to continue executing the test even if the assertion fails. In this case that'll probably result in other spurious error messages, because it's very unlikely for the rest of the test to succeed if dataset creation failed. So we probably don't want to use a subtest here.

Alternatively we could consider this part of the test setup, and assume that the dataset creation functionality is being thoroughly tested elsewhere. We can then drop any assertions related to the setup, simply assuming that it'll work.

The same goes for the "Dataset catalog is created" test below.